controllers/vpcendpoint/helpers.go

```diff
@@ -140,3 +143,50 @@ func (r *VpcEndpointReconciler) defaultResourceRecord(resource *v1alpha1.VpcEndp
 		Value: vpceResp.VpcEndpoints[0].DnsEntries[0].DnsName,
 	}, nil
 }
+
+func (r *VpcEndpointReconciler) ensureExternalNameService(ctx context.Context, resource *v1alpha1.VpcEndpoint) error {
+	externalNameSvcSpec := client.ObjectKey{
+		Namespace: resource.Spec.ExternalNameService.Namespace,
+		Name:      resource.Spec.ExternalNameService.Name,
+	}
+
+	err := r.Client.Get(ctx, externalNameSvcSpec, &corev1.Service{})
```

Comment:
A little unfamiliar with `client.IgnoreNotFound()`, but should this line instead be

`err := client.IgnoreNotFound(r.Client.Get(ctx, externalNameSvcSpec, &corev1.Service{}))` ?

This would allow us to eliminate an err != nil check